Mathlib/RingTheory/Localization/Module.lean

```diff
@@ -76,82 +148,35 @@ variable [hA : IsLocalization (Algebra.algebraMapSubmonoid A S) Aₛ]
 open Submodule
 
 theorem LinearIndependent.localization_localization {ι : Type*} {v : ι → A}
-    (hv : LinearIndependent R v) : LinearIndependent Rₛ (algebraMap A Aₛ ∘ v) := by
-  rw [linearIndependent_iff'] at hv ⊢
-  intro s g hg i hi
-  choose! a g' hg' using IsLocalization.exist_integer_multiples S s g
-  have h0 : algebraMap A Aₛ (∑ i in s, g' i • v i) = 0 := by
-    apply_fun ((a : R) • ·) at hg
-    rw [smul_zero, Finset.smul_sum] at hg
-    rw [map_sum, ← hg]
-    refine' Finset.sum_congr rfl fun i hi => _
-    rw [← smul_assoc, ← hg' i hi, Algebra.smul_def, map_mul, ← IsScalarTower.algebraMap_apply, ←
-      Algebra.smul_def, algebraMap_smul, Function.comp_apply]
-  obtain ⟨⟨_, r, hrS, rfl⟩, hr : algebraMap R A r * _ = 0⟩ :=
-    (IsLocalization.map_eq_zero_iff (Algebra.algebraMapSubmonoid A S) _ _).1 h0
-  simp_rw [Finset.mul_sum, ← Algebra.smul_def, smul_smul] at hr
-  specialize hv s _ hr i hi
-  rw [← (IsLocalization.map_units Rₛ a).mul_right_eq_zero, ← Algebra.smul_def, ← hg' i hi]
-  exact (IsLocalization.map_eq_zero_iff S _ _).2 ⟨⟨r, hrS⟩, hv⟩
+    (hv : LinearIndependent R v) : LinearIndependent Rₛ ((algebraMap A Aₛ) ∘ v) :=
+  hv.of_isLocalizedModule Rₛ S (IsScalarTower.toAlgHom R A Aₛ).toLinearMap
 #align linear_independent.localization_localization LinearIndependent.localization_localization
 
-theorem SpanEqTop.localization_localization {v : Set A} (hv : span R v = ⊤) :
-    span Rₛ (algebraMap A Aₛ '' v) = ⊤ := by
-  rw [eq_top_iff]
-  rintro a' -
-  obtain ⟨a, ⟨_, s, hs, rfl⟩, rfl⟩ :=
-    IsLocalization.mk'_surjective (Algebra.algebraMapSubmonoid A S) a'
-  rw [IsLocalization.mk'_eq_mul_mk'_one, mul_comm, ← map_one (algebraMap R A)]
-  erw [← IsLocalization.algebraMap_mk' A Rₛ Aₛ (1 : R) ⟨s, hs⟩]
-  -- `erw` needed to unify `⟨s, hs⟩`
-  rw [← Algebra.smul_def]
-  refine' smul_mem _ _ (span_subset_span R Rₛ _ _)
-  rw [← Algebra.coe_linearMap, ← LinearMap.coe_restrictScalars R, ← LinearMap.map_span]
-  exact mem_map_of_mem (hv.symm ▸ mem_top)
-#align span_eq_top.localization_localization SpanEqTop.localization_localization
+theorem span_eq_top_localization_localization {v : Set A} (hv : span R v = ⊤) :
+    span Rₛ (algebraMap A Aₛ '' v) = ⊤ :=
+  span_eq_top_of_isLocalizedModule Rₛ S (IsScalarTower.toAlgHom R A Aₛ).toLinearMap hv
+#align span_eq_top.localization_localization span_eq_top_localization_localization
 
 /-- If `A` has an `R`-basis, then localizing `A` at `S` has a basis over `R` localized at `S`.
 
 A suitable instance for `[Algebra A Aₛ]` is `localizationAlgebra`.
 -/
 noncomputable def Basis.localizationLocalization {ι : Type*} (b : Basis ι R A) : Basis ι Rₛ Aₛ :=
-  Basis.mk (b.linearIndependent.localization_localization _ S _)
-    (by rw [Set.range_comp, SpanEqTop.localization_localization Rₛ S Aₛ b.span_eq])
-#align basis.localization_localization Basis.localizationLocalization
```

Comment:
It seems that you have removed some align statements. Is this intentional?

Reply:
No. it seems I made a mess there. Thanks for the report